modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1206,6 +1221,25 @@ private void _validateActive(
 		}
 	}
 
+	private void _validateEnableCategorization(
+			boolean enableCategorization, boolean system)
+		throws PortalException {
+
+		if (enableCategorization && system) {
+			throw new ObjectDefinitionEnableCategorizationException(
+				"ObjectDefinition must not be system to enable categorization");
```

Comment:
I think that the word "ObjectDefinition" could be separated, because I searched and found this example in ObjectDefinitionNameException.java:
`super("System object definition names must not start with \"C_\"");`
or you could rewrite like: "System object definition must not enable categorization"

Reply:
I also found another example in ObjectFieldSettingNameException.java:
`"The settings %s are not allowed for object field %s",`
Maybe, we could do something "similar" like:
 "Enable categorization is not allowed for system object definition"
  "The setting categorization is not allowed for system object definition"